### PR TITLE
Improve the release process documentation (fixes #919)

### LINKED ISSUE #919: Improve the release process documentation

### PR DESCRIPTION
**Pull request by olim7t**
**What this PR does**:
Make the release process easier to follow and less error-prone. Most tasks can be expressed as copy-pastable commands.

**Which issue(s) this PR fixes**:
Fixes #919

**Checklist**
- [x] Changes manually tested
- [ ] Automated Tests added/updated
- [x] Documentation added/updated
- [x] CHANGELOG.md updated (not required for documentation PRs)
- [x] CLA Signed:  [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: docs/content/en/contribute/releasing/_index.md
Comment (olim7t) [inline docs/content/en/contribute/releasing/_index.md:89]: The only downside is that this loses blank lines and comments.
We currently have a couple of blank lines in `kustomization.yaml`, and no comments in either file.

File: docs/content/en/contribute/releasing/_index.md
Comment (olim7t) [inline docs/content/en/contribute/releasing/_index.md:136]: Are e2e tests supposed to pass at this stage? (the tag doesn't exist yet)
If not it should be documented.

**Review by adejanovski — APPROVED** [review]
Awesome improvements!
The last push of the procedure needs a fix for the release branch and the `diff --git` command wouldn't work for me.
You can merge this after fixing the push command and I'll leave it up to you to decide for the diff command.

**Pull request merged**
